R/ggridges.R

```diff
@@ -0,0 +1,289 @@
+#' Get data for ridge plots
+#'
+#' @param data dataframe, the data returned by `ggplot2::ggplot_build()`.
+#' @param na.rm boolean, from params
+#'
+#' @return dataframe containing plotting data 
+#'
+get_ridge_data <- function(data, na.rm) {
+    if (isTRUE(na.rm)) {
+        data <- data[stats::complete.cases(data[c("x", "ymin", "ymax")]), ]
+    }
+    
+    #if dataframe is empty there's nothing to draw
+    if (nrow(data) == 0) return(list())
+    
+    # remove all points that fall below the minimum height
+    data$ymax[data$height < data$min_height] <- NA
+    
+    # order data
+    data <- data[order(data$ymin, data$x), ]
+    
+    # remove missing points
+    missing_pos <- !stats::complete.cases(data[c("x", "ymin", "ymax")])
+    ids <- cumsum(missing_pos) + 1
+    data$group <- paste0(data$group, "-", ids)
+    data[!missing_pos, ]
+    
```

Comment:
I also prefer no whitespace between the last line and closing `}`

```suggestion
```

Reply:
done